### PR TITLE
Nautobot-version-aware docker compose environment and invoke tasks

### PR DESCRIPTION
**Pull request by glennmatthews**
# What's Changed

- Add capability in `tasks.py` to detect the current ~Git base branch (from `main`, `develop`, `next`, `ltm-1.6`)~ _Nautobot major/minor version number_ and update the project name accordingly
- Enhance docker-compose-using tasks to pass the ~Git base branch~ _Nautobot major/minor version number_ as an environment variable to docker compose
- Enhance docker compose YAML files to use different image labels depending on the provided ~Git base branch~ `$NAUTOBOT_VER`
- Add `invoke branch` task for switching between and/or creating new branches, ensuring that it automatically stops the current Docker containers before doing so.

The net result here **should** be that once we have this enhancement in all of the major base branches, switching between release trains (or child branches under them) should be able to automatically use a different Docker image and different db volume depending on the relevant major.minor version. This should eliminate the time-consuming need to `invoke destroy build` whenever switching between a bugfix branch and a feature branch or vice versa.

Please pull this down and try it out.

# Screenshots

```
❯ invoke build
...
#39 naming to docker.io/local/nautobot-dev:local-2.3-py3.12 done
```

```
❯ invoke start
Starting Nautobot in detached mode...
Running docker compose command "up --detach"
PYTHON_VER=3.12 \
NAUTOBOT_VERSION=2.3 \
docker compose \
    --project-name "nautobot-2-3" \
    --project-directory "/Users/glennmatthews/Documents/git-stuff/nautobot/development/" \
    -f "/Users/glennmatthews/Documents/git-stuff/nautobot/development/docker-compose.yml" \
    -f "/Users/glennmatthews/Documents/git-stuff/nautobot/development/docker-compose.postgres.yml" \
    -f "/Users/glennmatthews/Documents/git-stuff/nautobot/development/docker-compose.dev.yml" up \
    --detach
 Network nautobot-2-3_default  Creating
 Network nautobot-2-3_default  Created
 Volume "nautobot-2-3_pgdata_nautobot"  Creating
 Volume "nautobot-2-3_pgdata_nautobot"  Created
 Volume "nautobot-2-3_media_root"  Creating
 Volume "nautobot-2-3_media_root"  Created
 Container nautobot-2-3-db-1  Creating
 Container nautobot-2-3-redis-1  Creating
 Container nautobot-2-3-selenium-1  Creating
 Container nautobot-2-3-redis-1  Created
 Container nautobot-2-3-selenium-1  Created
 Container nautobot-2-3-db-1  Created
 Container nautobot-2-3-nautobot-1  Creating
 Container nautobot-2-3-nautobot-1  Created
...
```

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- n/a Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- n/a Example App Updates (when adding/changing features)
- [x] Prepare corresponding PR to `ltm-1.6`
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by glennmatthews** [comment]
Breaks CI at present because the `actions/checkout` action checks out a detached HEAD instead of just checking out the branch, so `get_current_git_base_branch` will need to account for that and handle it gracefully.

File: development/docker-compose.yml [hunk 1]
```diff
@@ -6,12 +6,13 @@ services:
     build:
       args:
         PYTHON_VER: "${PYTHON_VER}"
+        BASE_BRANCH: "${BASE_BRANCH}"
```
Comment (glennmatthews) [inline development/docker-compose.yml:9]: Probably not needed, not a Dockerfile arg.

File: tasks.py [hunk 2]
```diff
@@ -150,9 +170,10 @@ def docker_compose(context, command, **kwargs):
         command (str): Command string to append to the "docker compose ..." command, such as "build", "up", etc.
         **kwargs: Passed through to the context.run() call.
     """
+    CURRENT_GIT_BASE_BRANCH = get_current_git_base_branch(context)
     compose_command_tokens = [
         "docker compose",
-        f'--project-name "{context.nautobot.project_name}"',
+        f'--project-name "{context.nautobot.project_name}-{CURRENT_GIT_BASE_BRANCH}"',
```
Comment (glennmatthews) [inline tasks.py:176]: docker-compose doesn't like `ltm-1.6` as a part of the project name, so need to fix it up to have legal characters:

```suggestion
        f'--project-name "{context.nautobot.project_name}-{CURRENT_GIT_BASE_BRANCH.replace(".", "_")}"',
```

File: development/docker-compose.yml (hunk 1 above)
Comment (glennmatthews) [reply development/docker-compose.yml:9]:
```suggestion
```

**Review by HanlinMiao — COMMENTED** [review]
When I did `invoke branch --branch next`, `git show-branch` seems to throw an error if I have more than 26 remote-tracking branches. I wonder if this is something we need to account for or if I should just clean up my local branches lol.
Screenshot 2024-12-16 at 1 09 07 PM

**Comment by HanlinMiao** [comment]
I think it does make the command fail. As you can see I am still on the original branch instead of `next`
Screenshot 2024-12-16 at 2 01 15 PM

**Comment by glennmatthews** [comment]
Alternate approach, hopefully more resilient - use the Nautobot major/minor version number as the differentiator between environments, rather than the error-prone attempts at determining the base Git branch.

**Review by HanlinMiao — APPROVED** [review]
Tried it locally and it worked with no problem. LGTM!

**Pull request merged**